### ISSUE TITLE
Argon2 tuning slow (>400ms)

### ISSUE THREAD

**Issue by dewyatt** [randombit/botan#2217]
I'm wondering if this can be improved?

```cpp
#include <chrono>
#include <botan/pwdhash.h>

using namespace std::chrono;

int main() {
    const int MSEC = 50;
    const int OUT_LEN = 16;

    auto fam = Botan::PasswordHashFamily::create("Argon2id");
    steady_clock::time_point begin = steady_clock::now();
    fam->tune(OUT_LEN, std::chrono::milliseconds(MSEC));
    steady_clock::time_point end = steady_clock::now();

    unsigned long long msec = duration_cast<std::chrono::milliseconds>(end - begin).count();
    printf("msec: %lld\n", msec);
}
```

**Comment by randombit** [comment]
Definitely seems off

**Comment by randombit** [comment]
The problem is tuning is done with larger params (256 MB of memory). Reason was to ensure we are testing performance of main memory rather than L2/L3 cache, since otherwise the runtime will not scale as we expect once we scale memory past the L3 size. But m=256mb, p=1,t=1 takes too long, 300-500 ms.

No reason to use 256 MB here though, even Zen2 has only 16 MiB of L3 per core. Reducing to 36 Mib seems to still give acurrate results across a wide range of requested runtimes and your example now runs in ~35 ms instead of ~300 for me.

Interestingly, even though my processor has only 2 MB L2 + 8 MB L3, I noticed scaling problems if I used only 16 MB for tuning - lower runtimes were correct but `botan pbkdf_tune --algo=Argon2id --max-mem=4096 --check 600` reported more like 800 ms to run the selected parameters. I'm not sure why that would be. At around 28 MB for tuning, it corrected itself. So there is some other factor here besides just the L2/L3 size.
